yrs/src/id_set.rs

```diff
@@ -216,17 +254,12 @@ impl Default for IdRange {
 
 impl Encode for IdRange {
     fn encode<E: Encoder>(&self, encoder: &mut E) {
-        match self {
-            IdRange::Continuous(range) => {
-                encoder.write_var(1u32);
-                range.encode(encoder)
-            }
-            IdRange::Fragmented(ranges) => {
-                encoder.write_var(ranges.len() as u32);
-                for range in ranges.iter() {
-                    range.encode(encoder);
-                }
-            }
+        if self.is_squashed() {
+            self.encode_raw(encoder)
+        } else {
+            let mut clone = self.clone();
```

Comment:
The fix is basically - check if `DeleteSet` was squashed and if not, clone it and squash before encoding.